examples/tutorials/nb_python/asset_viewer.py

```diff
@@ -0,0 +1,972 @@
+    visual_sensor._sensor_object.set_transformation_from_spec()
+    # save ID of agent being modified
+    init_state["agent_ID"] = agent_ID
+    # save agent initial state
+    init_state["agent_state"] = sim.get_agent(agent_ID).get_state()
+    # boost the agent off the floor
+    sim.get_agent(agent_ID).scene_node.translation += np.array([0, 1.5, 0])
+    return init_state
+
+
+# Reset sensor and agent state using dictionary built in init_camera_track_config
+# Used at end of cell that directly modifies camera (i.e. tracking an object)
+def restore_camera_track_config(sim, init_state):
+    visual_sensor = init_state["visual_sensor"]
+    agent_ID = init_state["agent_ID"]
+    # reset the sensor state for other examples
+    visual_sensor._spec.position = init_state["position"]
+    visual_sensor._spec.orientation = init_state["orientation"]
+    visual_sensor._sensor_object.set_transformation_from_spec()
+    # restore the agent's state to what was savedd in init_camera_track_config
+    sim.get_agent(agent_ID).set_state(init_state["agent_state"])
+
+
+# Simulate scene while having camera track COM of objects of interest
+def camera_track_simulate(sim, objects, dt=2.0, get_frames=True, agent_ID=0):
+    start_time = sim.get_world_time()
+    observations = []
+    num_objs = len(objects)
+    if num_objs == 0:
+        print("camera_track_simulate : Aborting, no objects sent to track")
+        return observations
+    agent = sim.get_agent(agent_ID)
+    # define up vector for tracking calc
+    up_vec = np.array([0, 1.0, 0])
+    # partition to speed up
+    time_step = 1.0 / 60.0
+    # process for multiple objects
+    while sim.get_world_time() - start_time < dt:
+        sim.step_physics(time_step)
+        # set agent state to look at object
+        camera_position = agent.scene_node.translation
+        camera_look_at = objects[0].translation
+        for i in range(1, num_objs):
+            camera_look_at += objects[i].translation
+        camera_look_at /= len(objects)
+        agent.scene_node.rotation = mn.Quaternion.from_matrix(
+            mn.Matrix4.look_at(camera_position, camera_look_at, up_vec).rotation()  # up
+        )
+        if get_frames:
+            observations.append(sim.get_sensor_observations())
+
+    return observations
+
+
+# Set an object transform relative to the agent state
+def set_object_state_from_agent(
+    sim,
+    obj,
+    offset=np.array([0, 2.0, -1.5]),
+    orientation=mn.Quaternion(((0, 0, 0), 1)),
+):
+    agent_transform = sim.agents[0].scene_node.transformation_matrix()
+    ob_translation = agent_transform.transform_point(offset)
+    obj.translation = ob_translation
+    obj.rotation = orientation
+
+
+# %%
+# @title Define Visualization Utility Function { display-mode: "form" }
+# @markdown (double click to show code)
+# @markdown - display_sample
+
+# Change to do something like this maybe: https://stackoverflow.com/a/41432704
+def display_sample(
```

Comment:
We really need to modernize and fix this util at some point... At the very least standardize it somewher eso it's easy to refactor later.